### PR TITLE
Fix 'Ordered comparison between pointer and zero' errors

### PR DESCRIPTION
**Pull request by nohal**
preventing the build with clang on macOS

### CODE REVIEW

File: src/Polar.cpp [hunk 1]
```diff
@@ -181,7 +181,7 @@ Polar::Polar()
 
 #define MAX_WINDSPEEDS_IN_TABLE 200
 #define MESSAGE(S) (S + wxString(_T("\n")) + wxString::FromUTF8(filename) \
-                    + (line > 0 ? (_(" line ") + wxString::Format(_T("%d"), linenum)) : _T("")))
+                    + (line > (void *)0 ? (_(" line ") + wxString::Format(_T("%d"), linenum)) : _T("")))
```
Comment (did-g) [inline src/Polar.cpp]: Hi,
IMO it's a thinko or a copy and paste error, testing line doesn't make sense.

Should be linenum > 0

Regards
Didier

**Comment by nohal** [comment]
Yep, might well be. Sean will deal with it somehow for sure...

**Comment by seandepagnier** [comment]
what is the compiler error message?

**Comment by nohal** [comment]
8 error like
```
/Users/nohal/Documents/source/OpenCPN/plugins/weather_routing_pi_/src/Polar.cpp:204:9: error: ordered comparison between pointer and zero ('char *' and 'int')
        PARSE_ERROR(_("Failed to open."));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/nohal/Documents/source/OpenCPN/plugins/weather_routing_pi_/src/Polar.cpp:187:37: note: expanded from macro 'PARSE_ERROR'
                                  + MESSAGE(S); goto failed; } while (0)
                                    ^~~~~~~~~~
/Users/nohal/Documents/source/OpenCPN/plugins/weather_routing_pi_/src/Polar.cpp:184:29: note: expanded from macro 'MESSAGE'
                    + (line > 0 ? (_(" line ") + wxString::Format(_T("%d"), linenum)) : _T("")))
                       ~~~~ ^ ~
```

**Comment by seandepagnier** [comment]
nice work, I think clang has uncovered a minor bug

**Pull request closed without merging**